### ISSUE TITLE
Remove-AzADServicePrincipal and New-AzADServicePrincipal Bug

### ISSUE THREAD

**Issue by AbhilashMDoijode** [Azure/azure-powershell#11929]
## Description
The command  Remove-AzADServicePrincipal (-DisplayName, -ApplicationId, -objectId) is not removing the service principal from the portal,When i run the command in VS code or Powershell ISE it says removed service principle but it exists in azure portal. like Remove-AzADServicePrincipal have same issue with other commands like Remove-AzADSpCredential -DisplayName ServicePrincipalName 

## Steps to reproduce

```powershell

$spName= Get-AzADServicePrincipal -DisplayName "SPName"
$appId= $spName.ApplicationId
Remove-AzADServicePrincipal -ApplicationId $appId

```

## Environment data

```
Name                           Value
----                           -----
PSVersion                      7.0.1
PSEdition                      Core
GitCommitId                    7.0.1
OS                             Microsoft Windows 10.0.18362
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

## Module versions

```powershell
Directory: C:\Users\adoijode\Documents\PowerShell\Modules

ModuleType Version    PreRelease Name                                PSEdition ExportedCommands
---------- -------    ---------- ----                                --------- ----------------
Script     3.3.2                 Pscx                                Desk      {Add-PathVariable, Clear-MSMQueue, ConvertFrom-Base64, ConvertTo-Base64…}

    Directory: C:\program files\powershell\7\Modules

ModuleType Version    PreRelease Name                                PSEdition ExportedCommands
---------- -------    ---------- ----                                --------- ----------------
Manifest   7.0.0.0               CimCmdlets                          Core      {Get-CimAssociatedInstance, Get-CimClass, Get-CimInstance, Get-CimSession…}
Manifest   1.2.5                 Microsoft.PowerShell.Archive        Desk      {Compress-Archive, Expand-Archive}
Manifest   7.0.0.0               Microsoft.PowerShell.Diagnostics    Core      {Get-WinEvent, New-WinEvent, Get-Counter}
Manifest   7.0.0.0               Microsoft.PowerShell.Host           Core      {Start-Transcript, Stop-Transcript}
Manifest   7.0.0.0               Microsoft.PowerShell.Management     Core      {Add-Content, Clear-Content, Get-Clipboard, Set-Clipboard…}
Manifest   7.0.0.0               Microsoft.PowerShell.Security       Core      {Get-Acl, Set-Acl, Get-PfxCertificate, Get-Credential…}
Manifest   7.0.0.0               Microsoft.PowerShell.Utility        Core      {Export-Alias, Get-Alias, Import-Alias, New-Alias…}
Manifest   7.0.0.0               Microsoft.WSMan.Management          Core      {Disable-WSManCredSSP, Enable-WSManCredSSP, Get-WSManCredSSP, Set-WSManQuickConfig…}
Script     1.4.6                 PackageManagement                   Desk      {Find-Package, Get-Package, Get-PackageProvider, Get-PackageSource…}
Script     2.2.4                 PowerShellGet                       Desk      {Find-Command, Find-DSCResource, Find-Module, Find-RoleCapability…}
Script     2.0.5                 PSDesiredStateConfiguration         Core      {Configuration, New-DscChecksum, Get-DscResource, Invoke-DscResource}
Script     7.0.0.0               PSDiagnostics                       Core      {Disable-PSTrace, Disable-PSWSManCombinedTrace, Disable-WSManTrace, Enable-PSTrace…}
Script     2.0.0                 PSReadLine                          Desk      {Get-PSReadLineKeyHandler, Set-PSReadLineKeyHandler, Remove-PSReadLineKeyHandler, Get-PSReadLineOption…}
Binary     2.0.3                 ThreadJob                           Desk      Start-ThreadJob

    Directory: C:\Program Files\WindowsPowerShell\Modules

ModuleType Version    PreRelease Name                                PSEdition ExportedCommands
---------- -------    ---------- ----                                --------- ----------------
Script     3.7.0                 Az                                  Core,Desk
Script     1.7.4                 Az.Accounts                         Core,Desk {Disable-AzDataCollection, Disable-AzContextAutosave, Enable-AzDataCollection, Enable-AzContextAutosave…}
Script     1.1.1                 Az.Advisor                          Core,Desk {Get-AzAdvisorRecommendation, Enable-AzAdvisorRecommendation, Disable-AzAdvisorRecommendation, Get-AzAdvisorConfiguration…}
Script     1.0.3                 Az.Aks                              Core,Desk {Get-AzAks, New-AzAks, Remove-AzAks, Import-AzAksCredential…}
Script     1.1.2                 Az.AnalysisServices                 Core,Desk {Resume-AzAnalysisServicesServer, Suspend-AzAnalysisServicesServer, Get-AzAnalysisServicesServer, Remove-AzAnalysisServicesServer…}
Script     1.4.0                 Az.ApiManagement                    Core,Desk {Add-AzApiManagementApiToProduct, Add-AzApiManagementProductToGroup, Add-AzApiManagementRegion, Add-AzApiManagementUserToGroup…}
Script     1.0.3                 Az.ApplicationInsights              Core,Desk {Get-AzApplicationInsights, New-AzApplicationInsights, Remove-AzApplicationInsights, Set-AzApplicationInsightsPricingPlan…}
Script     1.3.6                 Az.Automation                       Core,Desk {Get-AzAutomationHybridWorkerGroup, Remove-AzAutomationHybridWorkerGroup, Get-AzAutomationJobOutputRecord, Import-AzAutomationDscNodeConfiguration…}
Script     2.0.2                 Az.Batch                            Core,Desk {Remove-AzBatchAccount, Get-AzBatchAccount, Get-AzBatchAccountKey, New-AzBatchAccount…}
Script     1.0.2                 Az.Billing                          Core,Desk {Get-AzBillingInvoice, Get-AzBillingPeriod, Get-AzEnrollmentAccount, Get-AzConsumptionBudget…}
Script     1.4.2                 Az.Cdn                              Core,Desk {Get-AzCdnProfile, Get-AzCdnProfileSsoUrl, New-AzCdnProfile, Remove-AzCdnProfile…}
Script     1.2.3                 Az.CognitiveServices                Core,Desk {Get-AzCognitiveServicesAccount, Get-AzCognitiveServicesAccountKey, Get-AzCognitiveServicesAccountSku, Get-AzCognitiveServicesAccountType…}
Script     3.6.0                 Az.Compute                          Core,Desk {Remove-AzAvailabilitySet, Get-AzAvailabilitySet, New-AzAvailabilitySet, Update-AzAvailabilitySet…}
Script     1.0.3                 Az.ContainerInstance                Core,Desk {New-AzContainerGroup, Get-AzContainerGroup, Remove-AzContainerGroup, Get-AzContainerInstanceLog}
Script     1.1.1                 Az.ContainerRegistry                Core,Desk {New-AzContainerRegistry, Get-AzContainerRegistry, Update-AzContainerRegistry, Remove-AzContainerRegistry…}
Script     1.1.0                 Az.DataBoxEdge                      Core,Desk {Get-AzDataBoxEdgeJob, Get-AzDataBoxEdgeDevice, Invoke-AzDataBoxEdgeDevice, New-AzDataBoxEdgeDevice…}
Script     1.7.0                 Az.DataFactory                      Core,Desk {Set-AzDataFactoryV2, Update-AzDataFactoryV2, Get-AzDataFactoryV2, Remove-AzDataFactoryV2…}
Script     1.0.2                 Az.DataLakeAnalytics                Core,Desk {Get-AzDataLakeAnalyticsDataSource, New-AzDataLakeAnalyticsCatalogCredential, Remove-AzDataLakeAnalyticsCatalogCredential, Set-AzDataLakeAnalyticsCatalogCredential…}
Script     1.2.8                 Az.DataLakeStore                    Core,Desk {Get-AzDataLakeStoreTrustedIdProvider, Remove-AzDataLakeStoreTrustedIdProvider, Remove-AzDataLakeStoreFirewallRule, Set-AzDataLakeStoreTrustedIdProvider…}
Script     1.1.0                 Az.DeploymentManager                Core,Desk {Get-AzDeploymentManagerArtifactSource, New-AzDeploymentManagerArtifactSource, Set-AzDeploymentManagerArtifactSource, Remove-AzDeploymentManagerArtifactSource…}
Script     1.0.2                 Az.DevTestLabs                      Core,Desk {Get-AzDtlAllowedVMSizesPolicy, Get-AzDtlAutoShutdownPolicy, Get-AzDtlAutoStartPolicy, Get-AzDtlVMsPerLabPolicy…}
Script     1.1.2                 Az.Dns                              Core,Desk {Get-AzDnsRecordSet, New-AzDnsRecordConfig, Remove-AzDnsRecordSet, Set-AzDnsRecordSet…}
Script     1.2.3                 Az.EventGrid                        Core,Desk {New-AzEventGridTopic, Get-AzEventGridTopic, Set-AzEventGridTopic, New-AzEventGridTopicKey…}
Script     1.4.3                 Az.EventHub                         Core,Desk {New-AzEventHubNamespace, Get-AzEventHubNamespace, Set-AzEventHubNamespace, Remove-AzEventHubNamespace…}
Script     1.4.0                 Az.FrontDoor                        Core,Desk {New-AzFrontDoor, Get-AzFrontDoor, Set-AzFrontDoor, Remove-AzFrontDoor…}
Script     3.1.0                 Az.HDInsight                        Core,Desk {Get-AzHDInsightJob, New-AzHDInsightSqoopJobDefinition, Wait-AzHDInsightJob, New-AzHDInsightStreamingMapReduceJobDefinition…}
Script     1.0.1                 Az.HealthcareApis                   Core,Desk {New-AzHealthcareApisService, Remove-AzHealthcareApisService, Set-AzHealthcareApisService, Get-AzHealthcareApisService}
Script     2.3.0                 Az.IotHub                           Core,Desk {Add-AzIotHubKey, Get-AzIotHubEventHubConsumerGroup, Get-AzIotHubConnectionString, Get-AzIotHubJob…}
Script     1.5.2                 Az.KeyVault                         Core,Desk {Add-AzKeyVaultCertificate, Update-AzKeyVaultCertificate, Stop-AzKeyVaultCertificateOperation, Get-AzKeyVaultCertificateOperation…}
Script     1.3.2                 Az.LogicApp                         Core,Desk {Get-AzIntegrationAccountAgreement, Get-AzIntegrationAccountAssembly, Get-AzIntegrationAccountBatchConfiguration, Get-AzIntegrationAccountCallbackUrl…}
Script     1.1.3                 Az.MachineLearning                  Core,Desk {Move-AzMlCommitmentAssociation, Get-AzMlCommitmentAssociation, Get-AzMlCommitmentPlanUsageHistory, Remove-AzMlCommitmentPlan…}
Script     1.0.2                 Az.ManagedServices                  Core,Desk {Get-AzManagedServicesAssignment, New-AzManagedServicesAssignment, Remove-AzManagedServicesAssignment, Get-AzManagedServicesDefinition…}
Script     1.0.2                 Az.MarketplaceOrdering              Core,Desk {Get-AzMarketplaceTerms, Set-AzMarketplaceTerms}
Script     1.1.1                 Az.Media                            Core,Desk {Sync-AzMediaServiceStorageKey, Set-AzMediaServiceKey, Get-AzMediaServiceKey, Get-AzMediaServiceNameAvailability…}
Script     1.6.2                 Az.Monitor                          Core,Desk {Get-AzMetricDefinition, Get-AzMetric, Remove-AzLogProfile, Get-AzLogProfile…}
Script     2.4.0                 Az.Network                          Core,Desk {Add-AzApplicationGatewayAuthenticationCertificate, Get-AzApplicationGatewayAuthenticationCertificate, New-AzApplicationGatewayAuthenticationCertificate, Remove-AzApplicationGatewayAuthenti… 
Script     1.1.1                 Az.NotificationHubs                 Core,Desk {Get-AzNotificationHub, Get-AzNotificationHubAuthorizationRule, Get-AzNotificationHubListKey, Get-AzNotificationHubPNSCredential…}
Script     1.3.4                 Az.OperationalInsights              Core,Desk {New-AzOperationalInsightsAzureActivityLogDataSource, New-AzOperationalInsightsCustomLogDataSource, Disable-AzOperationalInsightsLinuxCustomLogCollection, Disable-AzOperationalInsightsIISLo…
Script     1.2.1                 Az.PolicyInsights                   Core,Desk {Get-AzPolicyEvent, Get-AzPolicyState, Get-AzPolicyStateSummary, Get-AzPolicyRemediation…}
Script     1.1.1                 Az.PowerBIEmbedded                  Core,Desk {Remove-AzPowerBIWorkspaceCollection, Get-AzPowerBIWorkspaceCollection, Get-AzPowerBIWorkspaceCollectionAccessKey, Get-AzPowerBIWorkspace…}
Script     1.0.2                 Az.PrivateDns                       Core,Desk {Get-AzPrivateDnsZone, Remove-AzPrivateDnsZone, Set-AzPrivateDnsZone, New-AzPrivateDnsZone…}
Script     2.8.0                 Az.RecoveryServices                 Core,Desk {Get-AzRecoveryServicesBackupProperty, Get-AzRecoveryServicesVault, Get-AzRecoveryServicesVaultSettingsFile, New-AzRecoveryServicesVault…}
Script     1.2.1                 Az.RedisCache                       Core,Desk {Remove-AzRedisCachePatchSchedule, New-AzRedisCacheScheduleEntry, Get-AzRedisCachePatchSchedule, New-AzRedisCachePatchSchedule…}
Script     1.0.3                 Az.Relay                            Core,Desk {New-AzRelayNamespace, Get-AzRelayNamespace, Set-AzRelayNamespace, Remove-AzRelayNamespace…}
Script     1.13.0                Az.Resources                        Core,Desk {Get-AzProviderOperation, Remove-AzRoleAssignment, Get-AzRoleAssignment, New-AzRoleAssignment…}
Script     1.4.1                 Az.ServiceBus                       Core,Desk {New-AzServiceBusNamespace, Get-AzServiceBusNamespace, Set-AzServiceBusNamespace, Remove-AzServiceBusNamespace…}
Script     2.0.1                 Az.ServiceFabric                    Core,Desk {Add-AzServiceFabricClientCertificate, Add-AzServiceFabricClusterCertificate, Add-AzServiceFabricNode, Add-AzServiceFabricNodeType…}
Script     1.1.1                 Az.SignalR                          Core,Desk {New-AzSignalR, Get-AzSignalR, Get-AzSignalRKey, New-AzSignalRKey…}
Script     2.5.0                 Az.Sql                              Core,Desk {Get-AzSqlDatabaseTransparentDataEncryption, Get-AzSqlDatabaseTransparentDataEncryptionActivity, Set-AzSqlDatabaseTransparentDataEncryption, Get-AzSqlDatabaseUpgradeHint…}
Script     1.1.0                 Az.SqlVirtualMachine                Core,Desk {New-AzSqlVM, Get-AzSqlVM, Update-AzSqlVM, Remove-AzSqlVM…}
Script     1.13.0                Az.Storage                          Core,Desk {Get-AzStorageAccount, Get-AzStorageAccountKey, New-AzStorageAccount, New-AzStorageAccountKey…}
Script     1.2.3                 Az.StorageSync                      Core,Desk {Invoke-AzStorageSyncCompatibilityCheck, New-AzStorageSyncService, Get-AzStorageSyncService, Remove-AzStorageSyncService…}
Script     1.0.1                 Az.StreamAnalytics                  Core,Desk {Get-AzStreamAnalyticsFunction, Get-AzStreamAnalyticsDefaultFunctionDefinition, New-AzStreamAnalyticsFunction, Remove-AzStreamAnalyticsFunction…}
Script     1.0.0                 Az.Support                          Core,Desk {Get-AzSupportService, Get-AzSupportProblemClassification, Get-AzSupportTicket, Get-AzSupportTicketCommunication…}
Script     1.0.3                 Az.TrafficManager                   Core,Desk {Add-AzTrafficManagerCustomHeaderToEndpoint, Remove-AzTrafficManagerCustomHeaderFromEndpoint, Add-AzTrafficManagerCustomHeaderToProfile, Remove-AzTrafficManagerCustomHeaderFromProfile…}
Script     1.8.0                 Az.Websites                         Core,Desk {Get-AzAppServicePlan, Set-AzAppServicePlan, New-AzAppServicePlan, Remove-AzAppServicePlan…}
Script     1.0.1                 Microsoft.PowerShell.Operation.Val… Desk      {Get-OperationValidation, Invoke-OperationValidation}
Manifest   1.1.183.57            MSOnline                            Desk      
Binary     1.0.0.1               PackageManagement                   Desk      {Find-Package, Get-Package, Get-PackageProvider, Get-PackageSource…}
Script     3.4.0                 Pester                              Desk      {Describe, Context, It, Should…}
Script     1.0.0.1               PowerShellGet                       Desk      {Install-Module, Find-Module, Save-Module, Update-Module…}
Script     2.0.0      beta2      PSReadline                          Desk      {Get-PSReadLineKeyHandler, Set-PSReadLineKeyHandler, Remove-PSReadLineKeyHandler, Get-PSReadLineOption…}

    Directory: C:\Windows\system32\WindowsPowerShell\v1.0\Modules

ModuleType Version    PreRelease Name                                PSEdition ExportedCommands
---------- -------    ---------- ----                                --------- ----------------
Manifest   1.0.0.0               AppBackgroundTask                   Core,Desk {Disable-AppBackgroundTaskDiagnosticLog, Enable-AppBackgroundTaskDiagnosticLog, Set-AppBackgroundTaskResourcePolicy, Unregister-AppBackgroundTask…}
Manifest   2.0.1.0               Appx                                Core,Desk {Add-AppxPackage, Get-AppxPackage, Get-AppxPackageManifest, Remove-AppxPackage…}
Script     1.0.0.0               AssignedAccess                      Core,Desk {Clear-AssignedAccess, Get-AssignedAccess, Set-AssignedAccess}
Manifest   1.0.0.0               BitLocker                           Core,Desk {Unlock-BitLocker, Suspend-BitLocker, Resume-BitLocker, Remove-BitLockerKeyProtector…}
Manifest   1.0.0.0               BranchCache                         Core,Desk {Add-BCDataCacheExtension, Clear-BCCache, Disable-BC, Disable-BCDowngrading…}
Manifest   1.0                   ConfigDefender                      Core,Desk {Get-MpPreference, Set-MpPreference, Add-MpPreference, Remove-MpPreference…}
Manifest   1.0                   Defender                            Core,Desk {Get-MpPreference, Set-MpPreference, Add-MpPreference, Remove-MpPreference…}
Manifest   1.0.2.0               DeliveryOptimization                Core,Desk {Delete-DeliveryOptimizationCache, Get-DeliveryOptimizationStatus, Set-DeliveryOptimizationStatus, Get-DeliveryOptimizationPerfSnap…}
Manifest   1.0.0.0               DirectAccessClientComponents        Core,Desk {Disable-DAManualEntryPointSelection, Enable-DAManualEntryPointSelection, Get-DAClientExperienceConfiguration, Get-DAEntryPointTableItem…}
Script     3.0                   Dism                                Core,Desk {Add-AppxProvisionedPackage, Add-WindowsDriver, Add-WindowsCapability, Add-WindowsImage…}
Manifest   1.0.0.0               DnsClient                           Core,Desk {Resolve-DnsName, Clear-DnsClientCache, Get-DnsClient, Get-DnsClientCache…}
Manifest   1.0.0.0               EventTracingManagement              Core,Desk {Start-EtwTraceSession, New-EtwTraceSession, Get-EtwTraceSession, Update-EtwTraceSession…}
Manifest   2.0.0.0               International                       Core,Desk {Get-WinDefaultInputMethodOverride, Set-WinDefaultInputMethodOverride, Get-WinHomeLocation, Set-WinHomeLocation…}
Manifest   3.0.0.0               Microsoft.PowerShell.Diagnostics    Core,Desk {Get-WinEvent, Get-Counter, Import-Counter, Export-Counter…}
Manifest   1.0.0.0               Microsoft.PowerShell.LocalAccounts  Core,Desk {Add-LocalGroupMember, Disable-LocalUser, Enable-LocalUser, Get-LocalGroup…}
Manifest   3.1.0.0               Microsoft.PowerShell.Management     Core,Desk {Add-Content, Clear-Content, Clear-ItemProperty, Join-Path…}
Manifest   1.0                   MMAgent                             Core,Desk {Disable-MMAgent, Enable-MMAgent, Set-MMAgent, Get-MMAgent…}
Manifest   2.0.0.0               NetAdapter                          Core,Desk {Disable-NetAdapter, Disable-NetAdapterBinding, Disable-NetAdapterChecksumOffload, Disable-NetAdapterEncapsulatedPacketTaskOffload…}
Manifest   1.0.0.0               NetConnection                       Core,Desk {Get-NetConnectionProfile, Set-NetConnectionProfile}
Manifest   1.0.0.0               NetEventPacketCapture               Core,Desk {New-NetEventSession, Remove-NetEventSession, Get-NetEventSession, Set-NetEventSession…}
Manifest   2.0.0.0               NetLbfo                             Core,Desk {Add-NetLbfoTeamMember, Add-NetLbfoTeamNic, Get-NetLbfoTeam, Get-NetLbfoTeamMember…}
Manifest   1.0.0.0               NetNat                              Core,Desk {Get-NetNat, Get-NetNatExternalAddress, Get-NetNatStaticMapping, Get-NetNatSession…}
Manifest   2.0.0.0               NetQos                              Core,Desk {Get-NetQosPolicy, Set-NetQosPolicy, Remove-NetQosPolicy, New-NetQosPolicy}
Manifest   2.0.0.0               NetSecurity                         Core,Desk {Get-DAPolicyChange, New-NetIPsecAuthProposal, New-NetIPsecMainModeCryptoProposal, New-NetIPsecQuickModeCryptoProposal…}
Manifest   1.0.0.0               NetSwitchTeam                       Core,Desk {New-NetSwitchTeam, Remove-NetSwitchTeam, Get-NetSwitchTeam, Rename-NetSwitchTeam…}
Manifest   1.0.0.0               NetTCPIP                            Core,Desk {Get-NetIPAddress, Get-NetIPInterface, Get-NetIPv4Protocol, Get-NetIPv6Protocol…}
Manifest   1.0.0.0               NetworkConnectivityStatus           Core,Desk {Get-DAConnectionStatus, Get-NCSIPolicyConfiguration, Reset-NCSIPolicyConfiguration, Set-NCSIPolicyConfiguration}
Manifest   1.0.0.0               NetworkSwitchManager                Core,Desk {Disable-NetworkSwitchEthernetPort, Enable-NetworkSwitchEthernetPort, Get-NetworkSwitchEthernetPort, Remove-NetworkSwitchEthernetPortIPAddress…}
Manifest   1.0.0.0               NetworkTransition                   Core,Desk {Add-NetIPHttpsCertBinding, Disable-NetDnsTransitionConfiguration, Disable-NetIPHttpsProfile, Disable-NetNatTransitionConfiguration…}
Manifest   1.0.0.0               PcsvDevice                          Core,Desk {Get-PcsvDevice, Start-PcsvDevice, Stop-PcsvDevice, Restart-PcsvDevice…}
Manifest   1.0.0.0               PnpDevice                           Core,Desk {Get-PnpDevice, Get-PnpDeviceProperty, Enable-PnpDevice, Disable-PnpDevice}
Manifest   1.1                   PrintManagement                     Core,Desk {Add-Printer, Add-PrinterDriver, Add-PrinterPort, Get-PrintConfiguration…}
Binary     1.0.11                ProcessMitigations                  Core,Desk {Get-ProcessMitigation, Set-ProcessMitigation, ConvertTo-ProcessMitigationPolicy}
Manifest   1.0.0.0               ScheduledTasks                      Core,Desk {Get-ScheduledTask, Set-ScheduledTask, Register-ScheduledTask, Unregister-ScheduledTask…}
Manifest   2.0.0.0               SecureBoot                          Core,Desk {Confirm-SecureBootUEFI, Set-SecureBootUEFI, Get-SecureBootUEFI, Format-SecureBootUEFI…}
Manifest   2.0.0.0               SmbShare                            Core,Desk {Get-SmbShare, Remove-SmbShare, Set-SmbShare, Block-SmbShareAccess…}
Manifest   2.0.0.0               SmbWitness                          Core,Desk {Get-SmbWitnessClient, Move-SmbWitnessClient, gsmbw, msmbw…}
Manifest   1.0.0.0               StartLayout                         Core,Desk {Export-StartLayout, Import-StartLayout, Export-StartLayoutEdgeAssets, Get-StartApps}
Manifest   2.0.0.0               Storage                             Core,Desk {Add-InitiatorIdToMaskingSet, Add-PartitionAccessPath, Add-PhysicalDisk, Add-StorageFaultDomain…}
Manifest   1.0.0.0               TroubleshootingPack                 Core,Desk {Get-TroubleshootingPack, Invoke-TroubleshootingPack}
Manifest   2.0.0.0               TrustedPlatformModule               Core,Desk {Get-Tpm, Initialize-Tpm, Clear-Tpm, Unblock-Tpm…}
Binary     2.1.639.0             UEV                                 Core,Desk 
Manifest   2.0.0.0               VpnClient                           Core,Desk {Add-VpnConnection, Set-VpnConnection, Remove-VpnConnection, Get-VpnConnection…}
Manifest   1.0.0.0               Wdac                                Core,Desk {Get-OdbcDriver, Set-OdbcDriver, Get-OdbcDsn, Add-OdbcDsn…}
Manifest   2.0.0.0               Whea                                Core,Desk {Get-WheaMemoryPolicy, Set-WheaMemoryPolicy}
Manifest   1.0.0.0               WindowsDeveloperLicense             Core,Desk {Get-WindowsDeveloperLicense, Unregister-WindowsDeveloperLicense, Show-WindowsDeveloperLicenseRegistration}
Script     1.0                   WindowsErrorReporting               Core,Desk {Enable-WindowsErrorReporting, Disable-WindowsErrorReporting, Get-WindowsErrorReporting}
Manifest   1.0.0.0               WindowsSearch                       Core,Desk {Get-WindowsSearchSetting, Set-WindowsSearchSetting}
Manifest   1.0.0.0               WindowsUpdate                       Core,Desk Get-WindowsUpdateLog
Manifest   1.0.0.2               WindowsUpdateProvider               Core,Desk {Get-WUAVersion, Get-WULastInstallationDate, Get-WULastScanSuccessDate, Get-WUIsPendingReboot…}

    Directory: C:\Users\adoijode\.vscode\extensions\ms-vscode.powershell-2020.4.0\modules

ModuleType Version    PreRelease Name                                PSEdition ExportedCommands
---------- -------    ---------- ----                                --------- ----------------
Script     1.1.3                 Plaster                             Desk      {Invoke-Plaster, New-PlasterManifest, Get-PlasterTemplate, Test-PlasterManifest}
Binary     2.0.0                 PowerShellEditorServices            Desk      Start-EditorServices
Binary     0.2.0                 PowerShellEditorServices.VSCode     Desk      {New-VSCodeHtmlContentView, Show-VSCodeHtmlContentView, Close-VSCodeHtmlContentView, Set-VSCodeHtmlContentView…}
Script     2.1.0      beta1      PSReadLine                          Desk      {Get-PSReadLineKeyHandler, Set-PSReadLineKeyHandler, Remove-PSReadLineKeyHandler, Get-PSReadLineOption…}
Script     1.18.3                PSScriptAnalyzer                    Desk      {Get-ScriptAnalyzerRule, Invoke-ScriptAnalyzer, Invoke-Formatter}
```

## Debug output

```
DEBUG: 12:31:16 PM - NewAzureADServicePrincipalCommand begin processing with ParameterSet 'SimpleParameterSet'.
DEBUG: 12:31:16 PM - using account id 'adoijode@kagool.com'...
DEBUG: [Common.Authentication]: Authenticating using Account: 'adoijode@kagool.com', environment: 'AzureCloud', tenant: '82583032-c12a-4a39-a1b7-a28b18bccc03'
DEBUG: [Common.Authentication]: Authenticating using configuration values: Domain: '82583032-c12a-4a39-a1b7-a28b18bccc03', Endpoint: 'https://login.microsoftonline.com/', ClientId: '1950a258-227b-4e31-a9cf-717495945fc2', ClientRedirect: 'urn:ietf:wg:oauth:2.0:oob', ResourceClientUri: 'https://graph.windows.net/', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using context with Authority 'https://login.microsoftonline.com/82583032-c12a-4a39-a1b7-a28b18bccc03/', CorrelationId: '00000000-0000-0000-0000-000000000000', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using AdalConfiguration with Domain: '82583032-c12a-4a39-a1b7-a28b18bccc03', AdEndpoint: 'https://login.microsoftonline.com/', ClientId: '1950a258-227b-4e31-a9cf-717495945fc2', ClientRedirectUri: urn:ietf:wg:oauth:2.0:oob 
DEBUG: [ADAL]: Information: 2020-05-19T07:01:16.3825269Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...

DEBUG: [ADAL]: Information: 2020-05-19T07:01:16.3827068Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...

DEBUG: [ADAL]: Information: 2020-05-19T07:01:16.3828273Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: === Token Acquisition started:
        CacheType: null
        Authentication Target: User
        , Authority Host: login.microsoftonline.com

DEBUG: [ADAL]: Information: 2020-05-19T07:01:16.3829441Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: === Token Acquisition started:
        Authority: https://login.microsoftonline.com/82583032-c12a-4a39-a1b7-a28b18bccc03/
        Resource: https://graph.windows.net/
        ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
        CacheType: null
        Authentication Target: User


DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:16.3832087Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: Loading from cache.

DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:16.3832871Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: Loading from cache.

DEBUG: [ADAL]: Information: 2020-05-19T07:01:16.3856351Z: 00000000-0000-0000-0000-000000000000 - LoggerBase.cs: Deserialized 3 items to token cache.

DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:16.3858198Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: Looking up cache for a token...

DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:16.3858841Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: Looking up cache for a token...

DEBUG: [ADAL]: Information: 2020-05-19T07:01:16.3860433Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: An item matching the requested resource was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:16.3861043Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: An item matching the requested resource was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:16.3862962Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: 51.807145436666666 minutes left until token in cache expires

DEBUG: [ADAL]: Information: 2020-05-19T07:01:16.3863580Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: 51.807145436666666 minutes left until token in cache expires

DEBUG: [ADAL]: Information: 2020-05-19T07:01:16.3864135Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:16.3864668Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:16.3867043Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 5/19/2020 7:53:04 AM +00:00

DEBUG: [ADAL]: Information: 2020-05-19T07:01:16.3867662Z: 0ea9f119-4018-474a-860a-138a616a8741 - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 5/19/2020 7:53:04 AM +00:00Access Token Hash: UaL5m0xQcljJI3aUawJBv3r44t9+Wgw9ZWKhGuCk7dQ=
         User id: 713e544f-ca17-485b-ba01-00e497adca0f

DEBUG: [Common.Authentication]: Renewing Token with Type: 'Bearer', Expiry: '05/19/2020 07:53:04 +00:00', MultipleResource? 'True', Tenant: '82583032-c12a-4a39-a1b7-a28b18bccc03', UserId: 'adoijode@kagool.com'
DEBUG: [Common.Authentication]: User info for token DisplayId: 'adoijode@kagool.com', Name: Abhilash Doijode, IdProvider: 'https://sts.windows.net/82583032-c12a-4a39-a1b7-a28b18bccc03/', Uid: '713e544f-ca17-485b-ba01-00e497adca0f'
DEBUG: [Common.Authentication]: Checking token expiration, token expires '05/19/2020 07:53:04 +00:00' Comparing to '05/19/2020 07:01:16 +00:00' With threshold '00:05:00', calculated time until token expiry: '00:51:48.3709527'
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
POST

Absolute Uri:
https://graph.windows.net/82583032-c12a-4a39-a1b7-a28b18bccc03/applications?api-version=1.6

Headers:
x-ms-client-request-id        : fd7ccc87-5ca5-4932-970b-825100b8ad6f
Accept-Language               : en-US

Body:
{
  "availableToOtherTenants": false,
  "displayName": "whatever",
  "homepage": "http://whatever",
  "identifierUris": [
    "http://whatever"
  ],
  "passwordCredentials": [
    {
      "startDate": "2020-05-19T07:01:16.3812243Z",
      "endDate": "2021-05-19T07:01:16.3812243Z",
      "keyId": "fae4361e-316b-48bc-b69e-f4d9cae75b25",
      "value": "8ee8127d-bb8d-4313-892d-3d2b9d6a40b6"
    }
  ]
}


DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
BadRequest

Headers:
Cache-Control                 : no-cache
Pragma                        : no-cache
ocp-aad-diagnostics-server-name: D/r56k5GOGnWvF0XK7nHP+qPW/uWPUaGWjNAquEEhQQ=
request-id                    : 2a64e661-313f-4bb1-81ae-f6215e2832ad
client-request-id             : d45a2e08-33e0-46c5-b571-f4b7b88a4f28
x-ms-dirapi-data-contract-version: 1.6
ocp-aad-session-key           : aPkOX_QSUnueXx6aLLkgFvQJ9ch7VB-w4O7I_bnLi2G7VEtam3Rh0NPx0-2sX6Y-V72rVm-dgmYAGS7YwkD3ul-41hgoXJ4RNmF-cRUj57XqNd1814KtUDvNIpENSMI1Ag3sJKVCf4lfNYN6bgc4wbyEEAZivFrQAvxCE9sVTn0.wPdp-4DwgHhGGbn6T3YfloPi9VE7QSVBZqPNIgb5NOs
DataServiceVersion            : 3.0;
Strict-Transport-Security     : max-age=31536000; includeSubDomains
Access-Control-Allow-Origin   : *
X-AspNet-Version              : 4.0.30319
X-Powered-By                  : ASP.NET
Duration                      : 6178911
Date                          : Tue, 19 May 2020 07:01:22 GMT

Body:
{
  "odata.error": {
    "code": "Request_BadRequest",
    "message": {
      "lang": "en",
      "value": "Another object with the same value for property identifierUris already exists."
    },
    "requestId": "2a64e661-313f-4bb1-81ae-f6215e2832ad",
    "date": "2020-05-19T07:01:22",
    "values": [
      {
        "item": "PropertyName",
        "value": "identifierUris"
      },
      {
        "item": "PropertyErrorCode",
        "value": "ObjectConflict"
      }
    ]
  }
}


DEBUG: Caught exception, type: Microsoft.Azure.Graph.RBAC.Models.GraphErrorException
New-AzADServicePrincipal: 
Line |
   2 |  New-AzADServicePrincipal -DisplayName "whatever"
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Another object with the same value for property identifierUris already exists.
DEBUG: AzureQoSEvent: CommandName - New-AzADServicePrincipal; IsSuccess - False; Duration - 00:00:01.7057351;; Exception - System.Exception: Another object with the same value for property identifierUris already exists.;
DEBUG: Finish sending metric.
DEBUG: 12:31:19 PM - NewAzureADServicePrincipalCommand end processing.
DEBUG: 12:31:19 PM - GetAzureADServicePrincipalCommand begin processing with ParameterSet 'DisplayNameParameterSet'.
DEBUG: 12:31:19 PM - using account id 'adoijode@kagool.com'...
DEBUG: [Common.Authentication]: Authenticating using Account: 'adoijode@kagool.com', environment: 'AzureCloud', tenant: '82583032-c12a-4a39-a1b7-a28b18bccc03'
DEBUG: [Common.Authentication]: Authenticating using configuration values: Domain: '82583032-c12a-4a39-a1b7-a28b18bccc03', Endpoint: 'https://login.microsoftonline.com/', ClientId: '1950a258-227b-4e31-a9cf-717495945fc2', ClientRedirect: 'urn:ietf:wg:oauth:2.0:oob', ResourceClientUri: 'https://graph.windows.net/', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using context with Authority 'https://login.microsoftonline.com/82583032-c12a-4a39-a1b7-a28b18bccc03/', CorrelationId: '00000000-0000-0000-0000-000000000000', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using AdalConfiguration with Domain: '82583032-c12a-4a39-a1b7-a28b18bccc03', AdEndpoint: 'https://login.microsoftonline.com/', ClientId: '1950a258-227b-4e31-a9cf-717495945fc2', ClientRedirectUri: urn:ietf:wg:oauth:2.0:oob 
DEBUG: [ADAL]: Information: 2020-05-19T07:01:19.2236681Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...

DEBUG: [ADAL]: Information: 2020-05-19T07:01:19.2238315Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...

DEBUG: [ADAL]: Information: 2020-05-19T07:01:19.2239342Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: === Token Acquisition started:
        CacheType: null
        Authentication Target: User
        , Authority Host: login.microsoftonline.com

DEBUG: [ADAL]: Information: 2020-05-19T07:01:19.2240042Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: === Token Acquisition started:
        Authority: https://login.microsoftonline.com/82583032-c12a-4a39-a1b7-a28b18bccc03/
        Resource: https://graph.windows.net/
        ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
        CacheType: null
        Authentication Target: User


DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:19.2242441Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: Loading from cache.

DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:19.2243093Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: Loading from cache.

DEBUG: [ADAL]: Information: 2020-05-19T07:01:19.2291237Z: 00000000-0000-0000-0000-000000000000 - LoggerBase.cs: Deserialized 3 items to token cache.

DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:19.2293251Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: Looking up cache for a token...

DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:19.2293968Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: Looking up cache for a token...

DEBUG: [ADAL]: Information: 2020-05-19T07:01:19.2295733Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: An item matching the requested resource was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:19.2296423Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: An item matching the requested resource was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:19.2298719Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: 51.75975257333333 minutes left until token in cache expires

DEBUG: [ADAL]: Information: 2020-05-19T07:01:19.2299412Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: 51.75975257333333 minutes left until token in cache expires

DEBUG: [ADAL]: Information: 2020-05-19T07:01:19.2300075Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:19.2300693Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:19.2302931Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 5/19/2020 7:53:04 AM +00:00

DEBUG: [ADAL]: Information: 2020-05-19T07:01:19.2303642Z: 85c79bba-cb58-43cc-b304-52b1608918ab - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 5/19/2020 7:53:04 AM +00:00Access Token Hash: UaL5m0xQcljJI3aUawJBv3r44t9+Wgw9ZWKhGuCk7dQ=
         User id: 713e544f-ca17-485b-ba01-00e497adca0f

DEBUG: [Common.Authentication]: Renewing Token with Type: 'Bearer', Expiry: '05/19/2020 07:53:04 +00:00', MultipleResource? 'True', Tenant: '82583032-c12a-4a39-a1b7-a28b18bccc03', UserId: 'adoijode@kagool.com'
DEBUG: [Common.Authentication]: User info for token DisplayId: 'adoijode@kagool.com', Name: Abhilash Doijode, IdProvider: 'https://sts.windows.net/82583032-c12a-4a39-a1b7-a28b18bccc03/', Uid: '713e544f-ca17-485b-ba01-00e497adca0f'
DEBUG: [Common.Authentication]: Checking token expiration, token expires '05/19/2020 07:53:04 +00:00' Comparing to '05/19/2020 07:01:19 +00:00' With threshold '00:05:00', calculated time until token expiry: '00:51:45.4509753'
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://graph.windows.net/82583032-c12a-4a39-a1b7-a28b18bccc03/servicePrincipals?$filter=displayName eq 'whatever'&api-version=1.6

Headers:
x-ms-client-request-id        : 7047155e-ee5e-433e-95f9-75fb0f536090
Accept-Language               : en-US

Body:



DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
OK

Headers:
Cache-Control                 : no-cache
Pragma                        : no-cache
ocp-aad-diagnostics-server-name: bFwjpyh/wHMfljRE1GNjPgjYFltwWzSWevmUhGiJNRU=
request-id                    : e7920b9d-6620-466a-a4cf-af3295d635f6
client-request-id             : 5df15aa2-14aa-48d4-8bcb-ca8e44852080
x-ms-dirapi-data-contract-version: 1.6
ocp-aad-session-key           : zHUiaMQYShPgM05s13E_nLDS1E7bdJoWRiCK2nyabbquwAfI8D9tOm-cmpzldTHYD3UaKNmbPkvHq4escd273OeFbcX4W_LoWkdnzAnPavcCCMkGi3H7rcRDFg1jegymxf9G1vCTwgNdey3O7BPR21RSgNqp1UQO6LYZaimHx9U.3TWlu8XzK1R3EGqlq8nDfUJqfWOSmga4OHhusqXUYEE
DataServiceVersion            : 3.0;
Strict-Transport-Security     : max-age=31536000; includeSubDomains
Access-Control-Allow-Origin   : *
X-AspNet-Version              : 4.0.30319
X-Powered-By                  : ASP.NET
Duration                      : 2929082
Date                          : Tue, 19 May 2020 07:01:24 GMT

Body:
{
  "odata.metadata": "https://graph.windows.net/82583032-c12a-4a39-a1b7-a28b18bccc03/$metadata#directoryObjects",
  "value": []
}


DEBUG: AzureQoSEvent: CommandName - Get-AzADServicePrincipal; IsSuccess - True; Duration - 00:00:00.8061913;
DEBUG: Finish sending metric.
DEBUG: 12:31:20 PM - GetAzureADServicePrincipalCommand end processing.
DEBUG: 12:31:20 PM - GetAzureADServicePrincipalCommand begin processing with ParameterSet 'DisplayNameParameterSet'.
DEBUG: 12:31:20 PM - using account id 'adoijode@kagool.com'...
DEBUG: [Common.Authentication]: Authenticating using Account: 'adoijode@kagool.com', environment: 'AzureCloud', tenant: '82583032-c12a-4a39-a1b7-a28b18bccc03'
DEBUG: [Common.Authentication]: Authenticating using configuration values: Domain: '82583032-c12a-4a39-a1b7-a28b18bccc03', Endpoint: 'https://login.microsoftonline.com/', ClientId: '1950a258-227b-4e31-a9cf-717495945fc2', ClientRedirect: 'urn:ietf:wg:oauth:2.0:oob', ResourceClientUri: 'https://graph.windows.net/', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using context with Authority 'https://login.microsoftonline.com/82583032-c12a-4a39-a1b7-a28b18bccc03/', CorrelationId: '00000000-0000-0000-0000-000000000000', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using AdalConfiguration with Domain: '82583032-c12a-4a39-a1b7-a28b18bccc03', AdEndpoint: 'https://login.microsoftonline.com/', ClientId: '1950a258-227b-4e31-a9cf-717495945fc2', ClientRedirectUri: urn:ietf:wg:oauth:2.0:oob 
DEBUG: [ADAL]: Information: 2020-05-19T07:01:21.0085306Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...

DEBUG: [ADAL]: Information: 2020-05-19T07:01:21.0087032Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...

DEBUG: [ADAL]: Information: 2020-05-19T07:01:21.0088739Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: === Token Acquisition started:
        CacheType: null
        Authentication Target: User
        , Authority Host: login.microsoftonline.com

DEBUG: [ADAL]: Information: 2020-05-19T07:01:21.0089948Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: === Token Acquisition started:
        Authority: https://login.microsoftonline.com/82583032-c12a-4a39-a1b7-a28b18bccc03/
        Resource: https://graph.windows.net/
        ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
        CacheType: null
        Authentication Target: User


DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:21.0092764Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: Loading from cache.

DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:21.0093784Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: Loading from cache.

DEBUG: [ADAL]: Information: 2020-05-19T07:01:21.0284794Z: 00000000-0000-0000-0000-000000000000 - LoggerBase.cs: Deserialized 3 items to token cache.

DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:21.0287888Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: Looking up cache for a token...

DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:21.0289089Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: Looking up cache for a token...

DEBUG: [ADAL]: Information: 2020-05-19T07:01:21.0291458Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: An item matching the requested resource was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:21.0292632Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: An item matching the requested resource was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:21.0295088Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: 51.729758673333336 minutes left until token in cache expires

DEBUG: [ADAL]: Information: 2020-05-19T07:01:21.0296119Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: 51.729758673333336 minutes left until token in cache expires

DEBUG: [ADAL]: Information: 2020-05-19T07:01:21.0297151Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:21.0298289Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:21.0300976Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 5/19/2020 7:53:04 AM +00:00

DEBUG: [ADAL]: Information: 2020-05-19T07:01:21.0302033Z: 30ced9bb-104a-42a4-9cf5-6947f86cdedd - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 5/19/2020 7:53:04 AM +00:00Access Token Hash: UaL5m0xQcljJI3aUawJBv3r44t9+Wgw9ZWKhGuCk7dQ=
         User id: 713e544f-ca17-485b-ba01-00e497adca0f

DEBUG: [Common.Authentication]: Renewing Token with Type: 'Bearer', Expiry: '05/19/2020 07:53:04 +00:00', MultipleResource? 'True', Tenant: '82583032-c12a-4a39-a1b7-a28b18bccc03', UserId: 'adoijode@kagool.com'
DEBUG: [Common.Authentication]: User info for token DisplayId: 'adoijode@kagool.com', Name: Abhilash Doijode, IdProvider: 'https://sts.windows.net/82583032-c12a-4a39-a1b7-a28b18bccc03/', Uid: '713e544f-ca17-485b-ba01-00e497adca0f'
DEBUG: [Common.Authentication]: Checking token expiration, token expires '05/19/2020 07:53:04 +00:00' Comparing to '05/19/2020 07:01:21 +00:00' With threshold '00:05:00', calculated time until token expiry: '00:51:43.5665911'
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://graph.windows.net/82583032-c12a-4a39-a1b7-a28b18bccc03/servicePrincipals?$filter=displayName eq 'SP-Avid-Devlopment-Velocity-LogicApp'&api-version=1.6

Headers:
x-ms-client-request-id        : 11d3964a-8944-46ca-bbcb-74dcb9c9af57
Accept-Language               : en-US

Body:



DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
OK

Headers:
Cache-Control                 : no-cache
Pragma                        : no-cache
ocp-aad-diagnostics-server-name: NpXhnrof6UUIkXqY3q+5luIESnmPzCZ6dOMDLSGFhsk=
request-id                    : 3a5f40a3-751c-4e65-b811-14ba8dba13e4
client-request-id             : 75cb1691-e0c0-476d-9a77-f6bbce4461eb
x-ms-dirapi-data-contract-version: 1.6
ocp-aad-session-key           : 9Bh7BIb9IOBE0Yc25oJaIL4JOO2AsJCLxNh_2WL_Hc7Y3wtzNXgowwVcPGhW3QtLvKHgQBKxrOZqb1P489_enG-_nsZkZe6iA9IDF6JMWK3AGzcQ8NE9kvBifa2FLM9f21IN2B-g_UlnAYyfnZqcVvU8_ijDZgBgXNBLPCD-Opc.09m4yfxV-WKIN6CtOeiuIGSsgYJ5bRUMoPLAtkEfVr4
DataServiceVersion            : 3.0;
Strict-Transport-Security     : max-age=31536000; includeSubDomains
Access-Control-Allow-Origin   : *
X-AspNet-Version              : 4.0.30319
X-Powered-By                  : ASP.NET
Duration                      : 2977571
Date                          : Tue, 19 May 2020 07:01:25 GMT

Body:
{
  "odata.metadata": "https://graph.windows.net/82583032-c12a-4a39-a1b7-a28b18bccc03/$metadata#directoryObjects",
  "value": []
}


DEBUG: AzureQoSEvent: CommandName - Get-AzADServicePrincipal; IsSuccess - True; Duration - 00:00:00.9209531;
DEBUG: Finish sending metric.
DEBUG: 12:31:22 PM - GetAzureADServicePrincipalCommand end processing.
DEBUG: 12:31:22 PM - RemoveAzureADServicePrincipalCommand begin processing with ParameterSet 'DisplayNameParameterSet'.
DEBUG: 12:31:22 PM - using account id 'adoijode@kagool.com'...
DEBUG: [Common.Authentication]: Authenticating using Account: 'adoijode@kagool.com', environment: 'AzureCloud', tenant: '82583032-c12a-4a39-a1b7-a28b18bccc03'
DEBUG: [Common.Authentication]: Authenticating using configuration values: Domain: '82583032-c12a-4a39-a1b7-a28b18bccc03', Endpoint: 'https://login.microsoftonline.com/', ClientId: '1950a258-227b-4e31-a9cf-717495945fc2', ClientRedirect: 'urn:ietf:wg:oauth:2.0:oob', ResourceClientUri: 'https://graph.windows.net/', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using context with Authority 'https://login.microsoftonline.com/82583032-c12a-4a39-a1b7-a28b18bccc03/', CorrelationId: '00000000-0000-0000-0000-000000000000', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using AdalConfiguration with Domain: '82583032-c12a-4a39-a1b7-a28b18bccc03', AdEndpoint: 'https://login.microsoftonline.com/', ClientId: '1950a258-227b-4e31-a9cf-717495945fc2', ClientRedirectUri: urn:ietf:wg:oauth:2.0:oob 
DEBUG: [ADAL]: Information: 2020-05-19T07:01:22.9056897Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...

DEBUG: [ADAL]: Information: 2020-05-19T07:01:22.9058250Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...

DEBUG: [ADAL]: Information: 2020-05-19T07:01:22.9059617Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: === Token Acquisition started:
        CacheType: null
        Authentication Target: User
        , Authority Host: login.microsoftonline.com

DEBUG: [ADAL]: Information: 2020-05-19T07:01:22.9060631Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: === Token Acquisition started:
        Authority: https://login.microsoftonline.com/82583032-c12a-4a39-a1b7-a28b18bccc03/
        Resource: https://graph.windows.net/
        ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
        CacheType: null
        Authentication Target: User


DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:22.9063179Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: Loading from cache.

DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:22.9063888Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: Loading from cache.

DEBUG: [ADAL]: Information: 2020-05-19T07:01:22.9088181Z: 00000000-0000-0000-0000-000000000000 - LoggerBase.cs: Deserialized 3 items to token cache.

DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:22.9090585Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: Looking up cache for a token...

DEBUG: [ADAL]: Verbose: 2020-05-19T07:01:22.9091826Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: Looking up cache for a token...

DEBUG: [ADAL]: Information: 2020-05-19T07:01:22.9093660Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: An item matching the requested resource was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:22.9094358Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: An item matching the requested resource was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:22.9096313Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: 51.69842323666666 minutes left until token in cache expires

DEBUG: [ADAL]: Information: 2020-05-19T07:01:22.9096977Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: 51.69842323666666 minutes left until token in cache expires

DEBUG: [ADAL]: Information: 2020-05-19T07:01:22.9097701Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:22.9098381Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache

DEBUG: [ADAL]: Information: 2020-05-19T07:01:22.9100548Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 5/19/2020 7:53:04 AM +00:00

DEBUG: [ADAL]: Information: 2020-05-19T07:01:22.9101446Z: c68cb2e7-ea2a-4c8b-b373-73e0bede8693 - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 5/19/2020 7:53:04 AM +00:00Access Token Hash: UaL5m0xQcljJI3aUawJBv3r44t9+Wgw9ZWKhGuCk7dQ=
         User id: 713e544f-ca17-485b-ba01-00e497adca0f

DEBUG: [Common.Authentication]: Renewing Token with Type: 'Bearer', Expiry: '05/19/2020 07:53:04 +00:00', MultipleResource? 'True', Tenant: '82583032-c12a-4a39-a1b7-a28b18bccc03', UserId: 'adoijode@kagool.com'
DEBUG: [Common.Authentication]: User info for token DisplayId: 'adoijode@kagool.com', Name: Abhilash Doijode, IdProvider: 'https://sts.windows.net/82583032-c12a-4a39-a1b7-a28b18bccc03/', Uid: '713e544f-ca17-485b-ba01-00e497adca0f'
DEBUG: [Common.Authentication]: Checking token expiration, token expires '05/19/2020 07:53:04 +00:00' Comparing to '05/19/2020 07:01:22 +00:00' With threshold '00:05:00', calculated time until token expiry: '00:51:41.9024682'
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://graph.windows.net/82583032-c12a-4a39-a1b7-a28b18bccc03/servicePrincipals?$filter=displayName eq 'whatever'&api-version=1.6

Headers:
x-ms-client-request-id        : bf289761-c878-465c-a6d8-2f2304d8aa11
Accept-Language               : en-US

Body:



DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
OK

Headers:
Cache-Control                 : no-cache
Pragma                        : no-cache
ocp-aad-diagnostics-server-name: s4ClyWxOB6QM2ISqO5JGoxSJZGLDjm6EOb60TI/TyrU=
request-id                    : c918a440-059b-461a-a68b-8521954d66be
client-request-id             : dc1c9f9d-790f-4f7d-abbe-58614b0234ef
x-ms-dirapi-data-contract-version: 1.6
ocp-aad-session-key           : 5kaSMS-Al8eSSGu9Pcculi7GO4BC1PkqflM8qBYlzIfLOJQvLiI5bHiTaZfSg0ZkPgmr80hSr2yf2IEUKSHw5taP06CieuzbdZl6XGSlCyj-gb_j5TPxf9Fq5b1MSlU2zguQiIUwCysOydYGkErggBMiEwf50eEXgtp59TGNaFA.WfP83619qPkpgiA51Nl6FFBr63uDy28YqTsRnvicLfk
DataServiceVersion            : 3.0;
Strict-Transport-Security     : max-age=31536000; includeSubDomains
Access-Control-Allow-Origin   : *
X-AspNet-Version              : 4.0.30319
X-Powered-By                  : ASP.NET
Duration                      : 2918918
Date                          : Tue, 19 May 2020 07:01:27 GMT

Body:
{
  "odata.metadata": "https://graph.windows.net/82583032-c12a-4a39-a1b7-a28b18bccc03/$metadata#directoryObjects",
  "value": []
}


DEBUG: Caught exception, type: System.ArgumentException
Remove-AzADServicePrincipal: 
Line |
  10 |  Remove-AzADServicePrincipal -DisplayName "whatever"
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Could not find a service principal with the name .
DEBUG: AzureQoSEvent: CommandName - Remove-AzADServicePrincipal; IsSuccess - False; Duration - 00:00:00.9608299;; Exception - System.ArgumentException: Could not find a service principal with the name .
   at Microsoft.Azure.Commands.ActiveDirectory.ActiveDirectoryBaseCmdlet.HandleException(Exception exception)
   at Microsoft.Azure.Commands.ActiveDirectory.ActiveDirectoryBaseCmdlet.ExecutionBlock(Action execAction)
   at Microsoft.Azure.Commands.ActiveDirectory.RemoveAzureADServicePrincipalCommand.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.<>c__3`1.<ExecuteSynchronouslyOrAsJob>b__3_0(T c)
   at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cmdlet, Action`1 executor)
   at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cmdlet)
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord();
DEBUG: Finish sending metric.
DEBUG: 12:31:25 PM - RemoveAzureADServicePrincipalCommand end processing.
```

## Error output

```
DEBUG: 12:32:51 PM - ResolveError begin processing with ParameterSet 'AnyErrorParameterSet'.
DEBUG: 12:32:51 PM - using account id 'adoijode@kagool.com'...
WARNING: Breaking changes in the cmdlet 'Resolve-AzError' :
WARNING:  - The `Resolve-Error` alias will be removed in a future release.  Please change any scripts that use this alias to use `Resolve-AzError` instead.


WARNING: NOTE : Go to https://aka.ms/azps-changewarnings for steps to suppress this breaking change warning, and other information on breaking changes in Azure PowerShell.


   HistoryId: 10

Message        : Could not find a service principal with the name .
StackTrace     :    at Microsoft.Azure.Commands.ActiveDirectory.ActiveDirectoryBaseCmdlet.HandleException(Exception exception)
                    at Microsoft.Azure.Commands.ActiveDirectory.ActiveDirectoryBaseCmdlet.ExecutionBlock(Action execAction)
                    at Microsoft.Azure.Commands.ActiveDirectory.RemoveAzureADServicePrincipalCommand.ExecuteCmdlet()
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.<>c__3`1.<ExecuteSynchronouslyOrAsJob>b__3_0(T c)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cmdlet, Action`1 executor)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cmdlet)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
Exception      : System.ArgumentException
InvocationInfo : {Remove-AzADServicePrincipal}
Line           : Remove-AzADServicePrincipal -DisplayName "whatever"
Position       : At line:10 char:1
                 + Remove-AzADServicePrincipal -DisplayName "whatever"
                 + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 10

Message        : Another object with the same value for property identifierUris already exists.
StackTrace     :
Exception      : System.Exception
InvocationInfo : {New-AzADServicePrincipal}
Line           : New-AzADServicePrincipal -DisplayName "whatever"

Position       : At line:2 char:1
                 + New-AzADServicePrincipal -DisplayName "whatever"
                 + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 10


   HistoryId: 5

Message        : Another object with the same value for property identifierUris already exists.
StackTrace     :
Exception      : System.Exception
InvocationInfo : {New-AzADServicePrincipal}
Line           : New-AzADServicePrincipal -DisplayName "whatever"
Position       : At line:1 char:1
                 + New-AzADServicePrincipal -DisplayName "whatever"
                 + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 5


The Azure PowerShell team is listening, please let us know how we are doing: https://aka.ms/azpssurvey?Q_CHL=FEEDBACK.

DEBUG: AzureQoSEvent: CommandName - Resolve-AzError; IsSuccess - True; Duration - 00:00:00.3130908;
DEBUG: Finish sending metric.
DEBUG: 12:32:52 PM - ResolveError end processing.
```

**Comment by dingmeng-xue** [comment]
Thanks for reporting. We suspect there is latency between delete call and the object is really removed. Could you check whether the object is still there after a while such as 10 minutes?

**Comment by dingmeng-xue** [comment]
@AbhilashMDoijode, do you still have the issue?

**Comment by dingmeng-xue** [comment]
@AbhilashMDoijode, hope your issue was already disappeared. I'm closing this issue. Please contact us if you meet new issue.
